code/src/store/slices/userSlice.ts

```diff
@@ -103,5 +103,7 @@ export const useIsAuthLoadingSelector = () =>
   useAppSelector((store) => store.user.isLoading);
 export const useUserDetailsSelector = () =>
   useAppSelector((store) => store.user.userDetails);
+export const useIsShopManagerSelector = () =>
+  useAppSelector((store) => store.user.userDetails?.role === "ROLE_SHOP_MANAGER");
```

Comment:
The better solution will be either creating `useRoleSelector` with role argument or just using existing `useUserDetailsSelector` and destructuring role

Reply:
Thank you! Changed into useUserDetailsSelector